File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/tools/file_watchers/FileWatcherFilesToWatchSettings.java
package com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.common.FileType;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableFileWatcherFilesToWatchSettings.class)
public interface FileWatcherFilesToWatchSettings {

  Optional<FileType> fileType();

  Optional<String> scope();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/tools/file_watchers/FileWatcherFilesToWatchSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.alexandrecarlton.idea.settings.layout.common.FileType

data class FileWatcherFilesToWatchSettings(

    @JsonProperty("File type")
    val fileType: FileType? = null,

    @JsonProperty("Scope")
    val scope: String? = null
)
